--- /tmp/kp-0.9.0/setup.py	2024-01-20 16:06:48.000000000 +0100
+++ kp-0.9.0/setup.py	2026-05-08 12:52:01.609450110 +0200
@@ -44,10 +44,11 @@
                       '-DKOMPUTE_OPT_BUILD_PYTHON=ON',
                       '-DKOMPUTE_OPT_LOG_LEVEL=Off',
                       '-DKOMPUTE_OPT_USE_SPDLOG=Off',
-                      '-DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON'
+                      '-DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON',
                       '-DPYTHON_EXECUTABLE=' + sys.executable,
                       '-DPYTHON_INCLUDE_DIR=' + sysconfig.get_path('include'),
                       '-DPYTHON_LIBRARY=' + sysconfig.get_path('stdlib'),
+                      '-DCMAKE_POLICY_VERSION_MINIMUM=3.5',
         ]
 
         cfg = 'Debug' if self.debug else 'Release'
